Core/src/org/sleuthkit/autopsy/modules/hashdatabase/HashDbManager.java

```diff
@@ -786,48 +1031,340 @@ public boolean equals(Object obj) {
             }
-            if (this.sendIngestMessages != other.sendIngestMessages) {
+        }
+        
+        @Override
+        public void firePropertyChange(String propertyName, Object oldValue, Object newValue){
+            this.propertyChangeSupport.firePropertyChange(propertyName, oldValue, newValue);
+        }
+        
+        @Override
+        public String toString(){
+            return getHashSetName();
+        }
+        
+
+        @Override
+        public int hashCode() {
+            int code = 23;
+            code = 47 * code + Objects.hashCode(this.hashSetName);
+            code = 47 * code + Objects.hashCode(this.version);
+            code = 47 * code + Integer.hashCode(this.referenceSetID);
+            code = 47 * code + Objects.hashCode(this.knownFilesType);
+            return code;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == null) {
+                return false;
+            }
+            if (getClass() != obj.getClass()) {
+                return false;
+            }
+            final CentralRepoHashSet other = (CentralRepoHashSet) obj;
+            if (!Objects.equals(this.hashSetName, other.hashSetName)) {
+                return false;
+            }
+            if (!Objects.equals(this.version, other.version)) {
                 return false;
             }
             if (this.knownFilesType != other.knownFilesType) {
                 return false;
             }
             return true;
         }
-    }
-
+    }    
+    
     /**
      * Worker thread to make an index of a database
      */
     private class HashDbIndexer extends SwingWorker<Object, Void> {
 
         private ProgressHandle progress = null;
-        private HashDb hashDb = null;
+        private SleuthkitHashSet hashDb = null;
 
-        HashDbIndexer(HashDb hashDb) {
+        HashDbIndexer(SleuthkitHashSet hashDb) {
             this.hashDb = hashDb;
         }
 
-        ;
-
         @Override
         protected Object doInBackground() {
-            hashDb.indexing = true;
+            hashDb.setIndexing(true);
             progress = ProgressHandle.createHandle(
-                    NbBundle.getMessage(this.getClass(), "HashDbManager.progress.indexingHashSet", hashDb.hashSetName));
+                    NbBundle.getMessage(this.getClass(), "HashDbManager.progress.indexingHashSet", hashDb.getHashSetName()));
             progress.start();
             progress.switchToIndeterminate();
             try {
-                SleuthkitJNI.createLookupIndexForHashDatabase(hashDb.handle);
+                SleuthkitJNI.createLookupIndexForHashDatabase(hashDb.getHandle());
             } catch (TskCoreException ex) {
-                Logger.getLogger(HashDb.class.getName()).log(Level.SEVERE, "Error indexing hash database", ex); //NON-NLS
+                Logger.getLogger(SleuthkitHashSet.class.getName()).log(Level.SEVERE, "Error indexing hash database", ex); //NON-NLS
```

Comment:
I think it could help in debugging if we identify the database that experienced the error.